services/addons/images/geo_msg_query/geo_msg_query.py

```diff
@@ -12,43 +13,50 @@ def set_mongo_client(MONGO_DB_URI, MONGO_DB):
 
 
 def create_message(original_message, msg_type):
-    latitude = None
-    longitude = None
-    if msg_type == "Bsm":
-        longitude = original_message["payload"]["data"]["coreData"]["position"][
-            "longitude"
-        ]
-        latitude = original_message["payload"]["data"]["coreData"]["position"][
-            "latitude"
-        ]
-    elif msg_type == "Psm":
-        longitude = original_message["payload"]["data"]["position"]["longitude"]
-        latitude = original_message["payload"]["data"]["position"]["latitude"]
-    if latitude and longitude:
-        timestamp_str = original_message["metadata"]["odeReceivedAt"]
-        # Truncate the nanoseconds to microseconds
-        timestamp_str = timestamp_str[:26] + "Z"
-        new_message = {
-            "type": "Feature",
-            "geometry": {
-                "type": "Point",
-                "coordinates": [
-                    longitude,
-                    latitude,
-                ],
-            },
-            "properties": {
-                "id": original_message["metadata"]["originIp"],
-                "timestamp": datetime.strptime(
-                    timestamp_str,
-                    "%Y-%m-%dT%H:%M:%S.%fZ",
-                ),
-                "msg_type": msg_type,
-            },
-        }
-        return new_message
-    else:
-        logging.warn(f"create_message: Could not create a message for type: {msg_type}")
+    try:
+        latitude = None
+        longitude = None
+        if msg_type == "Bsm":
+            longitude = original_message["payload"]["data"]["coreData"]["position"][
+                "longitude"
+            ]
+            latitude = original_message["payload"]["data"]["coreData"]["position"][
+                "latitude"
+            ]
+        elif msg_type == "Psm":
+            longitude = original_message["payload"]["data"]["position"]["longitude"]
+            latitude = original_message["payload"]["data"]["position"]["latitude"]
+        if latitude and longitude:
+            timestamp_str = original_message["metadata"]["odeReceivedAt"]
+            if len(timestamp_str) > 26:
+                timestamp_str = timestamp_str[:26] + "Z"
```

Comment:
A couple comments would be useful here to explain why we are checking whether the timestamp string is longer than 26 characters and why it's necessary to truncate in that case.